server/server.go

```diff
@@ -307,7 +305,7 @@ func (s *Server) GetConfig(instID flux.InstanceID, includeSecrets bool) (flux.In
 
 	config := fullConfig.Settings
 	if !includeSecrets {
-		removeSecrets(&config)
+		config = config.HideSecrets()
```

Comment:
> cross fingers there ends up being no good reason to ever transmit secrets

Actually I don't think there is any good reason to ever send secrets back; and, having implemented the marshalling trick discussed above, I can report that it's just confusing.

So I think I will remove the secrets option, *always* hide secrets, and simply put the public key in the Key field (you can always generate the fingerprint if you want to check, or we can add an option for it).